bigtop-deploy/puppet/manifests/site.pp

```diff
@@ -24,8 +24,10 @@
 node default {
   $roles_enabled = hiera("bigtop::roles_enabled", false)
 
-  if (!is_bool($roles_enabled)) {
-    fail("bigtop::roles hiera conf is not of type boolean. It should be set to either true or false")
+  if ($operatingsystem != 'openEuler') {
```

Comment:
May I ask why we skip checking the `roles_enabled` statement for `openEuler` here?

Reply:
openeuler using ruby 3.0.3 version, the error log is:
![image](https://github.com/apache/bigtop/assets/11262916/8f7d49a4-12c3-4bab-bf97-058a97edaa68)

Reply:
> `is_bool()` is the function offered by `puppetlabs`. Why wasn't `is_bool()` found in opeEuler environment? Was the `puppet`installed correctly on OpenEuler?

is_bool() is the Ruby 3.x API
https://doc.wikimedia.org/puppet/puppet_functions_ruby3x/is_bool.html
![image](https://github.com/apache/bigtop/assets/11262916/bf69a0bd-c811-4238-816d-d1f75ccae2ea)
![image](https://github.com/apache/bigtop/assets/11262916/c04d4280-1367-449c-943a-fd6362fcc206)

Reply:
Would you please check puppet version in your deployed openeuler docker container?

In Rocky 8:
```
$ puppet --version
6.26.0
```

In Fedora-36:
```
$ puppet --version
7.20.0
```

Reply:
In openEuerl 22.03

```
$ puppet --version
7.22.0
```

Reply:
You should confirm whether puppetlabs was properly installed or not when deploying docker containers.
`is_bool` is the function of puppetlabs : https://github.com/puppetlabs/puppetlabs-stdlib/blob/v8.6.0/lib/puppet/parser/functions/is_bool.rb.

If `puppetlabs` was installed correctly, `is_bool` was called appropriately.

Reply:
https://github.com/apache/bigtop/pull/1132

Reply:
> #1132

There are 2 options:
1. Create the Jira and link it to #1132 (add Jira number to Pr title).
2. Or just add installation of puppetlabs-stdlib-4.12.0 into this PR.

Reply:
> > #1132
> 
> There are 2 options:
> 
> 1. Create the Jira and link it to [install puppetlabs-stdlib 4.12.0 by puppet in openeuler #1132](https://github.com/apache/bigtop/pull/1132) (add Jira number to Pr title).
> 2. Or just add installation of puppetlabs-stdlib-4.12.0 into this PR.

ok, i add  installation of puppetlabs-stdlib-4.12.0 into this PR:
commit: d006a324e422cef6217354fd0e99956a73f60ede